### PR TITLE
Track ALL devices for deviceWaitidle calls in ThreadSafety validation object

### LINKED ISSUE #2336: ThreadSafety::...DeviceWaitIdle should treat all queues as being written to
**Describe the situation in which you encountered the missing validation**
We got a bug report from a developer about a driver crash, which ended up being due to calling vkDeviceWaitIdle and vkQueueSubmit for one of that device's queues at the same time. ThreadSafety::{Pre,Post}CallRecordDeviceWaitIdle should call {Start,Finish}WriteObject on all queues belonging to the device.

**Valid Usage IDs requested**
UNASSIGNED-Threading-MultipleThreads

**Additional Context**
None

### PR DESCRIPTION
**Pull request by mark-lunarg**
Needed to guard all devices for these calls.

Fixes #2336.

### CODE REVIEW

File: scripts/thread_safety_generator.py [hunk 1]
```diff
@@ -1333,7 +1333,22 @@ class ThreadSafety : public ValidationObject {
 
 #endif // VK_USE_PLATFORM_XLIB_XRANDR_EXT
 
+void ThreadSafety::PreCallRecordDeviceWaitIdle(
+    VkDevice                                    device) {
+    const auto snapshot = parent_instance->c_VkDevice.object_table.snapshot();
```
Comment (jeffbolznv) [inline scripts/thread_safety_generator.py:1338]: It's not all devices that need to be tracked, it's all queues belonging to the passed-in device that need to be tracked.

Comment (mark-lunarg) [reply scripts/thread_safety_generator.py:1338]: Thanks, guess what little mind I have is already on vacation.  Will redo.

File: layers/generated/thread_safety.cpp [hunk 2]
```diff
@@ -719,7 +719,24 @@ void ThreadSafety::PostCallRecordGetRandROutputDisplayEXT(
 
 #endif // VK_USE_PLATFORM_XLIB_XRANDR_EXT
 
+void ThreadSafety::PreCallRecordDeviceWaitIdle(
+    VkDevice                                    device) {
+    StartReadObjectParentInstance(device, "vkDeviceWaitIdle");
+    const auto snapshot = c_VkQueue.object_table.snapshot();
```
Comment (jeffbolznv) [inline layers/generated/thread_safety.cpp:725]: c_vkQueue has all queues for all devices. I think you want device_queues_map[device] (and I guess a read_lock on thread_safety_lock to protect the access to the list?).

Comment (mark-lunarg) [reply layers/generated/thread_safety.cpp]: Updated.

File: layers/generated/thread_safety.cpp [hunk 3]
```diff
@@ -719,7 +719,26 @@ void ThreadSafety::PostCallRecordGetRandROutputDisplayEXT(
 
 #endif // VK_USE_PLATFORM_XLIB_XRANDR_EXT
 
+void ThreadSafety::PreCallRecordDeviceWaitIdle(
+    VkDevice                                    device) {
+    StartReadObjectParentInstance(device, "vkDeviceWaitIdle");
+    auto lock = write_lock_guard_t(thread_safety_lock);
```
Comment (jeffbolznv) [inline layers/generated/thread_safety.cpp:725]: These can be "read_lock_guard_t". Otherwise, the change LGTM.

**Comment by mark-lunarg** [comment]
Made write/read changes and pushed to Master.  Thanks @jeffbolznv for the detailed guidance and patience.

**Pull request closed without merging**